File: Merge Strings Alternately - Leetcode 1768/Merge Strings Alternately - Leetcode 1768.java
class Solution {
    public String mergeAlternately(String word1, String word2) {
        int w1Len = word1.length();
        int w2Len = word2.length();

        int length = Math.max(word1.length(), word2.length());

        String mergedString = "";
        for (int i = 0; i < length; i++) {
            if(w1Len>i){
                if(i == w1Len-1){
                    mergedString += word1.substring(i);
                }else{
                    mergedString += word1.substring(i,i+1);
                }
            }
            if(w2Len>i){
                if(i == w2Len-1){
                    mergedString += word2.substring(i);
                }else{
                    mergedString += word2.substring(i,i+1);
                }
            }
        }
        return mergedString;
    }
}

File: Merge Strings Alternately - Leetcode 1768/Merge Strings Alternately - Leetcode 1768.py
class Solution:
    def mergeAlternately(self, word1: str, word2: str) -> str:
        string = ""
        i = 0
        while ((len(word1) - i-1 >= 0) or (len(word2) - i-1>= 0)):
            if (len(word1)-i-1 >= 0):
                string += word1[i]
            if (len(word2)-i-1 >= 0):
                string += word2[i]
            i += 1
        return string